src/Stratis.Bitcoin/Consensus/ChainedHeaderTree.cs

```diff
@@ -250,6 +237,21 @@ public void Initialize(ChainedHeader consensusTip, bool blockStoreEnabled)
             this.logger.LogTrace("(-)");
         }
 
+        // <inheritdoc />
+        public ChainedHeaderBlock GetChainedHeaderBlock(uint256 blockHash)
+        {
+            this.logger.LogTrace("({0}:{1})", nameof(blockHash), blockHash);
+
+            ChainedHeaderBlock chainedHeaderBlock = null;
+            if (this.chainedHeadersByHash.TryGetValue(blockHash, out ChainedHeader chainedHeader))
+            {
+                chainedHeaderBlock = new ChainedHeaderBlock(chainedHeader.Block, chainedHeader);
+            }
+
+            this.logger.LogTrace("(-):'{0}'", chainedHeaderBlock);
```

Comment:
We don't have overload for to string here, we need to go with the chained header part.

Reply:
We have now

Reply:
oh, my mistake, okay then